### PR TITLE
Improvements to HTTP Middleware, Documentation and added Go Doc Examples

### PR DESCRIPTION
**Pull request by basvanbeek**
Various improvements and finetunings mostly dealing with documentation and http middleware.

### CODE REVIEW

File: middleware/http/client.go [hunk 1]
```diff
@@ -50,45 +59,49 @@ func TransportOptions(options ...TransportOption) ClientOption {
...
-	client.Transport = transport
+	c.Client.Transport = transport
 
 	return c, nil
 }
 
-// DoWithTrace wraps http.Client's Do with tracing using an application span.
-func (c *Client) DoWithTrace(req *http.Request, name string) (res *http.Response, err error) {
-	appSpan := c.tracer.StartSpan(name, zipkin.Kind(model.Client))
+// DoWithAppSpan wraps http.Client's Do with tracing using an application span.
```
Comment (adriancole) [inline middleware/http/client.go]: what does "application span" mean? At first I assumed it was about the span created, but then noticed the span created has kind.client

File: middleware/http/server_test.go [hunk 2]
```diff
@@ -101,3 +106,36 @@ func TestHTTPHandlerWrapping(t *testing.T) {
 		t.Errorf("Expected body value %q, got %q", want, have)
 	}
 }
+
+func TestHTTPMethodAsSpanName(t *testing.T) {
```
Comment (adriancole) [inline middleware/http/server_test.go]: what part of this implies the span name is the method (or is it that method is default?) If the latter, maybe name test to say defaultmethodisspanname?

File: middleware/http/client.go (hunk 1 above)
Comment (basvanbeek) [reply middleware/http/client.go]: yes that's a left over error... I will remove the zipkin.Kind

File: middleware/http/server_test.go (hunk 2 above)
Comment (basvanbeek) [reply middleware/http/server_test.go]: If Span name is not provided as functional option indeed the HTTP Method will be used.

Don't like `defaultmethodisspanname` either.

How about: TestHTTPDefaultSpanName?

**Comment by adriancole** [comment]
>
> How about: TestHTTPDefaultSpanName?
>
sgtm!

**Pull request merged**